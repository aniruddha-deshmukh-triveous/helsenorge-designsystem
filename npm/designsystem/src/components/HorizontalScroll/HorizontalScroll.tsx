import React, { useRef } from 'react';
import classNames from 'classnames';
import { useIsVisible } from '../../hooks/useIsVisible';
import styles from './styles.module.scss';

// Scrolle-indikator vises/skjules når det er x px igjen til venstre eller høyre side
const ROOT_MARGIN_OFFSET = '3px';

export const HorizontalScroll: React.FC = ({ children }) => {
  const viewportRef = useRef<HTMLDivElement>(null);
  const leftRef = useRef<HTMLDivElement>(null);
  const rightRef = useRef<HTMLDivElement>(null);
  const leftIsVisible = useIsVisible(leftRef, 1, { root: viewportRef?.current, rootMargin: ROOT_MARGIN_OFFSET }, true);
  const rightIsVisible = useIsVisible(rightRef, 1, { root: viewportRef?.current, rootMargin: ROOT_MARGIN_OFFSET }, true);

  return (
    <div className={styles.horizontalscroll}>
      <div className={styles.horizontalscroll__viewport} ref={viewportRef}>
        <div
          className={classNames(
            styles.horizontalscroll__indicator,
            styles['horizontalscroll__indicator--left'],
            !leftIsVisible && styles['horizontalscroll__indicator--visible']
          )}
        />
        <div
          className={classNames(
            styles.horizontalscroll__indicator,
            styles['horizontalscroll__indicator--right'],
            !rightIsVisible && styles['horizontalscroll__indicator--visible']
          )}
        />
        <div ref={leftRef} />
        {children}
        <div ref={rightRef} />
      </div>
    </div>
  );
};

export default HorizontalScroll;
