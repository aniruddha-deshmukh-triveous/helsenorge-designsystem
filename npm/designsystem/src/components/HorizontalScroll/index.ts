import HorizontalScroll from './HorizontalScroll';
export * from './HorizontalScroll';
export default HorizontalScroll;
