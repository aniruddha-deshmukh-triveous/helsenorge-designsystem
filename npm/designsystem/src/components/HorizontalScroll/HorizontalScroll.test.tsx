import React from 'react';

import { render, screen } from '@testing-library/react';

import HorizontalScroll from './HorizontalScroll';

describe('Gitt at HorizontalScroll skal rendres', (): void => {
  describe('Når HorizontalScroll skal vises vanlig', (): void => {
    test('Så rendres HorizontalScroll riktig', (): void => {
      const { container } = render(<HorizontalScroll>{'Teksten sin'}</HorizontalScroll>);

      const text = screen.getByText('Teksten sin');
      expect(text).toBeVisible();

      expect(container).toMatchSnapshot();
    });
  });
});
