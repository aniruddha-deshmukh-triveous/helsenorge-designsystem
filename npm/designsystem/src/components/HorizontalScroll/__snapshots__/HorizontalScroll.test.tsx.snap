// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at HorizontalScroll skal rendres Når HorizontalScroll skal vises vanlig Så rendres HorizontalScroll riktig 1`] = `
<div>
  <div
    class="horizontalscroll"
  >
    <div
      class="horizontalscroll__viewport"
      tabindex="0"
    >
      <div
        class="horizontalscroll__indicator horizontalscroll__indicator--left"
      />
      <div
        class="horizontalscroll__indicator horizontalscroll__indicator--right"
      />
      <div />
      Teksten sin
      <div />
    </div>
  </div>
</div>
`;
